tests/ArmTemplates.Tests/Creator/TemplateCreatorTests/TagTemplateCreatorTests.cs

```diff
@@ -10,28 +10,150 @@
+            tagTemplate.Parameters.Count().Should().Be(1);
+            tagTemplate.Parameters.ContainsKey(ParameterNames.ApimServiceName).Should().BeTrue();
+
+            tagTemplate.Resources.Count().Should().Be(5);
+            
+            foreach (TagTemplateResource tag in tagTemplate.Resources)
+            {
+                var resourceTagName = tag.Name.Split("/")[1].Split("'")[0];
+                expectedTagDictionary.Should().ContainKey(resourceTagName);
+                var generatedTagDisplayName = expectedTagDictionary[resourceTagName];
+                tag.Properties.DisplayName.Equals(generatedTagDisplayName).Should().BeTrue();
+            }
+        }
+
+        [Fact]
+        public void CreateTagTemplate_ShouldCreateTemplateFromCreatorConfig_GivenOnlyApiTags()
+        {
+            var tagTemplateCreator = new TagTemplateCreator(new TemplateBuilder());
+
+            var apiTagNames = new List<string>()
+            {
+                "tag 1", "tag2", "api tag 3", "api tag2"
+            };
+            var creatorConfig = this.GenerateCreatorParameters(apiTagNames: apiTagNames);
+
+            var expectedTagsDictionary = new Dictionary<string, string>()
+            {
+                { "tag-1", "tag 1" },
+                { "tag2", "tag2" },
+                { "api-tag-3", "api tag 3" },
+                { "api-tag2", "api tag2" }
+            };
+
+            //act
+            var tagTemplate = tagTemplateCreator.CreateTagTemplate(creatorConfig);
+
+            //assert
+            tagTemplate.Resources.Count().Should().Be(4);
+
+            foreach (TagTemplateResource tag in tagTemplate.Resources)
+            {
+                var resourceTagName = tag.Name.Split("/")[1].Split("'")[0];
+                expectedTagsDictionary.Should().ContainKey(resourceTagName);
+                var generatedTagDisplayName = expectedTagsDictionary[resourceTagName];
+                tag.Properties.DisplayName.Equals(generatedTagDisplayName).Should().BeTrue();
+            }
+        }
+
+        [Fact]
+        public void CreateTagTemplate_ShouldThrowDuplicateTagResourceNameException_GivenDuplicateSanitizedDisplayName()
+        {
+            var tagTemplateCreator = new TagTemplateCreator(new TemplateBuilder());
+
+            var tagNames= new List<string>()
+            {
+                "tag 1?", "?tag 1?"
+            };
+
+            var creatorConfig = this.GenerateCreatorParameters(tagNames: tagNames);
+
+            foreach (var tagName in tagNames)
+            {
+                var tag = new TagProperties
+                {
+                    DisplayName = tagName
+                };
+                creatorConfig.Tags.Add(tag);
+            }
+
+            //act & assert
+            var exception = Assert.Throws<DuplicateTagResourceNameException>(() => tagTemplateCreator.CreateTagTemplate(creatorConfig));
```

Comment:
we have `FluentAssertions` in `resource-kit`. This allows to right fancy `.Should()` methods to assert something.
And they have an implementation for checking exceptions: [ShouldThrow()](https://fluentassertions.com/exceptions/)

Reply:
Thank you, updated